components/edit_post_modal/edit_post_modal.test.tsx

```diff
@@ -0,0 +1,863 @@
+
+const defaultPost: Post = {
+    id: '123',
+    create_at: 1620568159640,
+    update_at: -1,
+    edit_at: -1,
+    delete_at: -1,
+    is_pinned: false,
+    user_id: '1',
+    channel_id: '5',
+    root_id: '123',
+    parent_id: '123',
+    original_id: '123',
+    message: 'test',
+    type: 'system_add_remove',
+    props: {},
+    hashtags: 'hashtag',
+    pending_post_id: '123',
+    reply_count: 0,
+    metadata: {
+        embeds: [],
+        emojis: [],
+        files: [],
+        images: {},
+        reactions: [],
+    },
+};
+
+const defaultProps = {
+    intl: jest.genMockFromModule<IntlShape>('react-intl'),
+    canEditPost: true,
+    canDeletePost: true,
+    useChannelMentions: true,
+    shouldShowPreview: true,
+    maxPostSize: 100,
+    config: {EnableEmojiPicker: 'true', EnableGifPicker: 'false'},
+    editingPost: {
+        postId: '123',
+        post: {
+            ...defaultPost,
+        },
+        commentCount: 3,
+        refocusId: 'test',
+        show: true,
+        title: 'test',
+    },
+    actions: {
+        editPost: jest.fn((data) => (Promise.resolve(data))),
+        addMessageIntoHistory: jest.fn(),
+        hideEditPostModal: jest.fn(),
+        openModal: jest.fn(),
+        setShowPreview: jest.fn(),
+    },
+};
+
+function createEditPost(
+    {
+        canEditPost = true,
+        canDeletePost = true,
+        useChannelMentions = true,
+        ctrlSend = undefined,
+        config = {EnableEmojiPicker: 'true', EnableGifPicker: 'true'},
+        editingPost = {post: {...defaultPost}, show: false, postId: '1'},
+        actions = {
+            editPost: jest.fn((data) => (Promise.resolve(data))),
+            addMessageIntoHistory: jest.fn(),
+            hideEditPostModal: jest.fn(),
+            openModal: jest.fn(),
+            setShowPreview: jest.fn(),
+        },
+    }: Props,
+) {
+    return (
+        <EditPostModal
+            canEditPost={canEditPost}
+            shouldShowPreview={false}
+            canDeletePost={canDeletePost}
+            ctrlSend={ctrlSend || false}
+            config={config}
+            editingPost={editingPost}
+            actions={actions}
+            maxPostSize={Constants.DEFAULT_CHARACTER_LIMIT}
+            useChannelMentions={useChannelMentions}
+        />
+    );
+}
+
+describe('components/EditPostModal', () => {
+    it('should match with default config', () => {
+        const wrapper = shallowWithIntl(createEditPost({...defaultProps}));
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should match without emoji picker', () => {
+        const config = {
+            PostEditTimeLimit: 300,
+            EnableEmojiPicker: 'false',
+        };
+        const wrapper = shallowWithIntl(createEditPost({...defaultProps, config}));
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should not call openModal on empty edited message but with attachment', () => {
+        window.scrollTo = () => {};
```

Comment:
What does adding this do?

Reply:
Without this, the tests fail with:
```
 Unexpected console logsError: Not implemented: window.scrollTo

      278 |         const data = await actions.editPost(updatedPost);
      279 |         if (data) {
    > 280 |             window.scrollTo(0, 0);
          |                    ^
      281 |         }
      282 |
      283 |         this.handleHide();

      at module.exports (node_modules/jsdom/lib/jsdom/browser/not-implemented.js:9:17)
      at scrollTo (node_modules/jsdom/lib/jsdom/browser/Window.js:728:7)
      at EditPostModal.handleEdit (components/edit_post_modal/edit_post_modal.tsx:280:20)
          at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:97:5),
      at Object.<anonymous> (tests/setup.js:67:15)
          at runMicrotasks (<anonymous>)
```

Reply:
Can we replace that line with this:
`global.scrollTo = jest.fn();`

We use this format in other parts of the test